tests/src/kernel/hipLaunchParm.cpp

```diff
@@ -23,75 +23,768 @@ THE SOFTWARE.
+  int li;  // local int
+  float lf;  // local float
+  bool result;  // local bool
+} hipLaunchKernelStruct_t1;
+
+// This test is to verify struct with padding, read from device
+typedef struct hipLaunchKernelStruct2 {
+  char c1;
+  long l1;
+  char c2;
+  long l2;
+  bool result;
+} hipLaunchKernelStruct_t2;
+
+// This test is to verify struct with padding, read from device
+typedef struct hipLaunchKernelStruct3 {
+  char bf1;
+  char bf2;
+  long l1;
+  char bf3;
+  bool result;
+} hipLaunchKernelStruct_t3;
+
+// This test is to verify empty struct
+typedef struct hipLaunchKernelStruct4 {
+  // empty struct, size will be verified from device side,size 1Byte
+} hipLaunchKernelStruct_t4;
+
+// This test is to verify struct with pointer member variable.
+typedef struct hipLaunchKernelStruct5 {
+  char c1;
+  char* cp;  // char pointer
+} hipLaunchKernelStruct_t5;
+
+
+// This test is to verify struct with aligned(8),
+// right now it's broken on hcc & hip-clang
+typedef struct hipLaunchKernelStruct6 {
+  char c1;
+  short int si;
+} __attribute__((aligned(8)))  hipLaunchKernelStruct_t6;
+
+// This test is to verify struct with aligned(16),
+// right now it's brokenon hcc & hip-clang
+typedef struct hipLaunchKernelStruct7 {
+  char c1;
+  short int si;
+} __attribute__((aligned(16)))  hipLaunchKernelStruct_t7;
+
+// This test is to verify struct with packed & aligned,
+// size should be 4Bytes right now it's broken on hcc & hip-clang
+typedef struct hipLaunchKernelStruct8 {
+  char c1;
+  short int si;
+  bool b;
+}__attribute__((packed, aligned(4)))  hipLaunchKernelStruct_t8;
+
+// This test is to verify const struct object
+typedef struct hipLaunchKernelStruct9 {
+  char c1;
+  uint32_t* ip;  // uint pointer
+} hipLaunchKernelStruct_t9;
+
+// This test is to verify struct with stdint types, uintN_t
+typedef struct hipLaunchKernelStruct10 {
+  uint64_t u64;
+  uint32_t u32;
+  uint8_t u8;
+} hipLaunchKernelStruct_t10;
+
+// This test is to verify struct with volatile member
+typedef struct hipLaunchKernelStruct11 {
+  int i1;
+  volatile unsigned int vint;
+} hipLaunchKernelStruct_t11;
+
+// This test is to verify struct with simple class object
+class base {
+ public:
+    int i = 0;
+    base() {}
+};
+typedef struct hipLaunchKernelStruct12 {
+  base b;
+  char c1;
+} hipLaunchKernelStruct_t12;
+
+// This test is to verify struct with __device__ func() attribute
+typedef struct hipLaunchKernelStruct13 {
+  int i1;
+  __device__ int getvalue() { return i1; }
+} hipLaunchKernelStruct_t13;
+
+// This test is to verify struct with array variable,
+// write to from device
+typedef struct hipLaunchKernelStruct14 {
+  int readint;
+  int writeint[BLOCK_DIM_SIZE];  // will write to this from device
+} hipLaunchKernelStruct_t14;
+
+// This test is to verify struct with dynamic memory, new int
+// the heap memory will be accessed from device
+typedef struct hipLaunchKernelStruct15 {
+  char c1;
+  int* heapmem = new int[BLOCK_DIM_SIZE];
```

Comment:
device code cannot access host heap memory. This memory has to be allocated by hipMalloc.

Reply:
Done